file_test.go

```diff
@@ -20,53 +20,50 @@ import (
 	"runtime"
 	"testing"
 
-	. "github.com/smartystreets/goconvey/convey"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"gopkg.in/ini.v1"
 )
 
 func TestEmpty(t *testing.T) {
-	Convey("Create an empty object", t, func() {
-		f := ini.Empty()
-		So(f, ShouldNotBeNil)
+	f := ini.Empty()
+	require.NotNil(t, f)
 
-		// Should only have the default section
-		So(len(f.Sections()), ShouldEqual, 1)
+	// Should only have the default section
+	assert.Len(t, f.Sections(), 1)
 
-		// Default section should not contain any key
-		So(len(f.Section("").Keys()), ShouldBeZeroValue)
-	})
+	// Default section should not contain any key
+	assert.Len(t, f.Section("").Keys(), 0)
 }
 
 func TestFile_NewSection(t *testing.T) {
-	Convey("Create a new section", t, func() {
-		f := ini.Empty()
-		So(f, ShouldNotBeNil)
+	f := ini.Empty()
+	require.NotNil(t, f)
 
-		sec, err := f.NewSection("author")
-		So(err, ShouldBeNil)
-		So(sec, ShouldNotBeNil)
-		So(sec.Name(), ShouldEqual, "author")
+	sec, err := f.NewSection("author")
+	require.NoError(t, err)
+	require.NotNil(t, sec)
+	assert.Equal(t, "author", sec.Name())
 
-		So(f.SectionStrings(), ShouldResemble, []string{ini.DefaultSection, "author"})
+	assert.Equal(t, []string{ini.DefaultSection, "author"}, f.SectionStrings())
 
-		Convey("With duplicated name", func() {
-			sec, err := f.NewSection("author")
-			So(err, ShouldBeNil)
-			So(sec, ShouldNotBeNil)
+	t.Run("with duplicated name", func(t *testing.T) {
+		sec, err := f.NewSection("author")
+		require.NoError(t, err)
+		require.NotNil(t, sec)
 
-			// Does nothing if section already exists
-			So(f.SectionStrings(), ShouldResemble, []string{ini.DefaultSection, "author"})
-		})
+		// Does nothing if section already exists
+		assert.Equal(t, []string{ini.DefaultSection, "author"}, f.SectionStrings())
+	})
 
-		Convey("With empty string", func() {
-			_, err := f.NewSection("")
-			So(err, ShouldNotBeNil)
-		})
+	t.Run("with empty string", func(t *testing.T) {
+		_, err := f.NewSection("")
+		require.Error(t, err)
```

Comment:
Pretty sure we may use `assert` at all times?

Reply:
That will allow the test to continue which is not a good idea if there is an error or a return value that is later used is nil (will lead to panics in the tests).

What do you want me to do?

Reply:
Ah! TIL. Yeah in that case agree to use `require`!